admin/cluster_containers_setup.py

```diff
@@ -0,0 +1,369 @@
+        """
+        Build a Flocker deployment configuration for the given cluster
+        and parameters.
+        The configuration consists of identically configured applications
+        (containers) uniformly spread over all cluster nodes.
+
+        :return dict: containing the json we need to send to compose to
+            create the datasets and containers we want.
+        """
+        Message.log(action="Building config")
+        application_root = {}
+        applications = {}
+        application_root["version"] = 1
+        application_root["applications"] = applications
+        for node in self.nodes:
+            for i in range(self.per_node):
+                name = "app_%s_%d" % (node.public_address, i)
+                applications[name] = deepcopy(self.application_template)
+
+        deployment_root = {}
+        nodes = {}
+        deployment_root["nodes"] = nodes
+        deployment_root["version"] = 1
+        for node in self.nodes:
+            addr = "%s" % node.public_address
+            nodes[addr] = []
+            for i in range(self.per_node):
+                name = "app_%s_%d" % (node.public_address, i)
+                nodes[addr].append(name)
+
+        return {"applications": application_root,
+                "deployment": deployment_root}
+
+    def _configure(self, configuration):
+        """
+        Configure the cluster with the given deployment configuration.
+
+        :param dict configuration: dictionary with the configuration
+            to deploy.
+        :return Deferred: Deferred that fires when the configuration is pushed
+                          to the cluster's control agent.
+        """
+        base_url = b"https://{}:{}/v1".format(
+            self.control_node_address, REST_API_PORT
+        )
+        cluster_cert = self.cluster_cert
+        user_cert = self.user_cert
+        user_key = self.user_key
+        body = dumps(configuration)
+        treq_client = treq_with_authentication(
+            self.reactor, cluster_cert, user_cert, user_key)
+
+        def do_configure():
+            posted = treq_client.post(
+                base_url + b"/configuration/_compose", data=body,
+                headers={b"content-type": b"application/json"},
+                persistent=False
+            )
+
+            def got_response(response):
+                if response.code != OK:
+                    d = json_content(response)
+
+                    def got_error(error):
+                        if isinstance(error, dict):
+                            error = error[u"description"] + u"\n"
+                        else:
+                            error = u"Unknown error: " + unicode(error) + "\n"
+                        raise ResponseError(response.code, error)
+
+                    d.addCallback(got_error)
+                    return d
+
+            posted.addCallback(got_response)
+            return posted
+
+        return do_configure()
+
```

Comment:
Remove the extra blank line at the end to avoid a lint error. You should have one blank line at the end of the file.